mesonbuild/backend/ninjabackend.py

```diff
@@ -199,7 +199,9 @@ def strToCommandArg(c: T.Union[NinjaCommandArg, str]) -> NinjaCommandArg:
                 # shell constructs shouldn't be shell quoted
                 return NinjaCommandArg(c, Quoting.notShell)
             if c.startswith('$'):
-                var = re.search(r'\$\{?(\w*)\}?', c).group(1)
+                varp = re.search(r'\$\{?(\w*)\}?', c)
+                assert varp is not None
```

Comment:
I guess this assert is for mypy, isn't it?

Reply:
Yeah, effectively. Although it just moves a potential crash to an assertion failure.

Reply:
Sure. But it's a case that "can't ever happen", right? So the assert makes sense.